src/main/java/bio/terra/service/dataset/flight/ingest/IngestValidateScratchTableStep.java

```diff
@@ -0,0 +1,64 @@
+package bio.terra.service.dataset.flight.ingest;
+
+import bio.terra.model.IngestRequestModel;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.dataset.DatasetTable;
+import bio.terra.service.dataset.exception.IngestFailureException;
+import bio.terra.service.filedata.azure.AzureSynapsePdao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import java.sql.SQLException;
+import java.util.List;
+
+public class IngestValidateScratchTableStep implements Step {
+
+  private AzureSynapsePdao azureSynapsePdao;
+  private DatasetService datasetService;
+
+  public IngestValidateScratchTableStep(
+      AzureSynapsePdao azureSynapsePdao, DatasetService datasetService) {
+    this.azureSynapsePdao = azureSynapsePdao;
+    this.datasetService = datasetService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException {
+    FlightMap workingMap = context.getWorkingMap();
+    IngestRequestModel ingestRequestModel = IngestUtils.getIngestRequestModel(context);
+
+    Dataset dataset = IngestUtils.getDataset(context, datasetService);
+    DatasetTable targetTable = IngestUtils.getDatasetTable(context, dataset);
+    int failCount;
+    try {
+      failCount =
+          azureSynapsePdao.validateScratchParquetFiles(
+              targetTable, IngestUtils.getSynapseScratchTableName(context.getFlightId()));
+
+    } catch (SQLException ex) {
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL, ex);
+    }
+
+    if (failCount > ingestRequestModel.getMaxBadRecords()) {
```

Comment:
It looks like when a user ingests data into a table with max bad records > 0, all of the following ingests to that table will fail unless they also set max bad records. Is that how it behaves in bigquery as well?

Reply:
Oh nevermind! I see we're just validating the scratch table...which is the whole point 🙂